port_ocean/core/handlers/entity_processor/jq_entity_processor.py

```diff
@@ -124,32 +135,22 @@ async def _calculate_entities(
                     raw_entity_mappings,
                     mapping.port.items_to_parse,
                     mapping.selector.query,
+                    parse_all,
                 )
             )
-            for data in raw_data
+            for data in raw_results
         ]
         entities = await asyncio.gather(*entities_tasks)
 
-        return [
-            Entity.parse_obj(entity_data)
-            for flatten in entities
-            for entity_data in filter(
-                lambda entity: entity.get("identifier") and entity.get("blueprint"),
-                flatten,
-            )
-        ]
-
-    async def _parse_items(
-        self, mapping: ResourceConfig, raw_results: RawEntityDiff
-    ) -> EntityDiff:
-        entities_before: list[Entity] = await self._calculate_entities(
-            mapping, raw_results["before"]
-        )
-        entities_after: list[Entity] = await self._calculate_entities(
-            mapping, raw_results["after"]
-        )
-
-        return {
-            "before": entities_before,
-            "after": entities_after,
-        }
+        passed = []
+        failed = []
+        for flatten in entities:
+            for entity_data, did_entity_pass_selector in flatten:
```

Comment:
```suggestion
        for entity_result in entities_results:
            for entity, did_entity_pass_selector in entity_result:
```

Reply:
btw I don't think this unpacking of tuple will work as `flatten` isn't an iterator. 
I think `entity_result[0]` `entity_result[1]` will suit better

Reply:
i'll make sure but flatten is a list of tuple so it should work

Reply:
yeah it works